api/controllers/calendarEntry.controller.ts

```diff
@@ -85,9 +85,26 @@ export const getCalendarEntry = async (
 export const deleteCalendarEntry = async (
   req: Request,
   res: Response,
-  _next: NextFunction
+  _next: NextFunction,
 ) => {
   const { id } = req.params;
   await CalendarEntry.deleteOne({ _id: id });
   res.status(200).json();
 };
+
+export const updateCalendarEntry = async (
+  req: Request,
+  res: Response,
+  _next: NextFunction,
+) => {
+  const { id } = req.params;
+  try {
+    const entry = await CalendarEntry.findByIdAndUpdate(
+      id,
+      req.body as CalendarEntry,
+    );
+    res.status(200).json(entry);
+  } catch {
+    res.status(400).json({ error: "Something went wrong" });
+  }
```

Comment:
It could be nice to use the error that is caught and log it, or use it in the returned error message. In this case, I think keeping it simple is ok, esp since I'll be adding more robust error handling.

Reply:
Ok, we will leave it for now.